tethysext/atcore/tests/integrated_tests/models/files_database/file_collection_client_tests.py

```diff
@@ -463,3 +474,327 @@ def test_bad_collection_client(self):
+        )
+        collection_client = FileCollectionClient(self.session, self.general_collection_id)
+        with self.assertRaises(FileCollectionItemNotFoundError) as exc:
+            collection_client.delete_item('file2.txt')
+        self.assertTrue('"file2.txt" not found in this collection.' in str(exc.exception))
+
+    def test_delete_item_directory(self):
+        """Test deleting an item directory."""
+        test_dir_name = 'test_delete_item_directory'
+        base_files_root_dir = os.path.join(self.test_files_base, test_dir_name)
+        root_dir = os.path.join(self.test_files_base, 'temp', test_dir_name)
+        self.copy_files_to_temp_directory(base_files_root_dir, root_dir)
+        _, collection_instance = self.get_database_and_collection(
+            database_id=self.general_database_id, collection_id=self.general_collection_id,
+            root_directory=root_dir, database_meta={}, collection_meta={}
+        )
+        collection_client = FileCollectionClient(self.session, self.general_collection_id)
+        collection_client.delete_item('dir1')
+        self.assertFalse(os.path.exists(os.path.join(collection_client.path, 'dir1')))
+
+    def test_delete_item_directory_does_not_exist(self):
+        """Test deleting a directory item that doesn't exist throws the correct error."""
+        test_dir_name = 'test_delete_item_directory_does_not_exist'
+        base_files_root_dir = os.path.join(self.test_files_base, test_dir_name)
+        root_dir = os.path.join(self.test_files_base, 'temp', test_dir_name)
+        self.copy_files_to_temp_directory(base_files_root_dir, root_dir)
+        _, collection_instance = self.get_database_and_collection(
+            database_id=self.general_database_id, collection_id=self.general_collection_id,
+            root_directory=root_dir, database_meta={}, collection_meta={}
+        )
+        collection_client = FileCollectionClient(self.session, self.general_collection_id)
+        with self.assertRaises(FileCollectionItemNotFoundError) as exc:
+            collection_client.delete_item('dir1')
+        self.assertTrue('"dir1" not found in this collection.' in str(exc.exception))
+
+    def test_open_file_read(self):
+        """Test opening a file for reading."""
+        test_dir_name = 'test_open_file_read'
+        base_files_root_dir = os.path.join(self.test_files_base, test_dir_name)
+        root_dir = os.path.join(self.test_files_base, 'temp', test_dir_name)
+        self.copy_files_to_temp_directory(base_files_root_dir, root_dir)
+        _, collection_instance = self.get_database_and_collection(
+            database_id=self.general_database_id, collection_id=self.general_collection_id,
+            root_directory=root_dir, database_meta={}, collection_meta={}
+        )
+        collection_client = FileCollectionClient(self.session, self.general_collection_id)
+        with collection_client.open_file('file1.txt', 'r') as f:
```

Comment:
Can `open_file` be used like:

```
f = collection_client.open_file('file1.txt', 'r')
content = f.read()
f.close()
```

If so, let's add tests for that usage as well. If not, I don't care that much. I use the context managed version of the `open` function most of the time anyway.

Reply:
Right now it doesn't work that way. That is why there isn't a test for that.